R/parse.R

```diff
@@ -205,15 +208,13 @@ ensure_valid_pd <- function(pd) {
 #' @param pd A parse table.
 #' @param text The initial code to style.
 #' @keywords internal
-identify_insufficiently_parsed_strings <- function(pd, text) {
-  is_problematic_string <- pd$token == "STR_CONST"
-  candidate_substring <- substr(
-    pd$text[is_problematic_string], 1, 1
-  )
-  is_problematic_string[is_problematic_string] <- candidate_substring == "["
-  is_problematic_string
+is_insufficiently_parsed_string <- function(pd) {
+  grepl("^\\[", pd$text) & pd$token == "STR_CONST"
 }
 
+is_insufficiently_parsed_number <- function(pd) {
+  grepl("^0x", pd$text)
```

Comment:
Do we need to check for `"STR_NUM"` here?

Reply:
I first hoped we don't but since parent matches children too, it messes things up.